hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -307,21 +314,21 @@ spec = around showLogsOnFailure $ do
                   seedFromFaucet_ node bobCardanoVk 100_000_000 Fuel (contramap FromFaucet tracer)
 
                   send n1 $ input "Init" []
-                  waitForAllMatch 10 [n1] $ headIsInitializingWith (Set.fromList [alice])
+                  [headIdAliceOnly] <- waitForAllMatch 10 [n1] $ headIsInitializingWith (Set.fromList [alice])
 
                   -- Bob opens and immediately aborts a Head with Alice, iow pulls Alice in
                   -- "his" Head
                   send n2 $ input "Init" []
-                  waitForAllMatch 10 [n2] $ headIsInitializingWith (Set.fromList [alice, bob])
+                  [headIdAliceAndBob] <- waitForAllMatch 10 [n2] $ headIsInitializingWith (Set.fromList [alice, bob])
```

Comment:
We could use `waitMatch` here and above (which only takes a single client as argument)